components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.admin.apis/AdminApiHelper/src/main/java/org/wso2/carbon/micro/integrator/admin/apis/AdminApiHelper.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.carbon.micro.integrator.admin.apis;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.util.AXIOMUtil;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.core.axis2.ProxyService;
+import org.apache.synapse.mediators.AbstractMediator;
+import org.apache.synapse.rest.API;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * Helper class to retrieve details of the proxy-services and API's running on the runtime.
+ */
+public class AdminApiHelper extends AbstractMediator {
```

Comment:
At the same time the helper class is a mediator. Can this be a helper class in that case? Can you brief how this fit to design?

Reply:
This class is used to retrieve details of the proxy-services and API's running on the runtime. We have an API which uses this to expose those details out side.